code/Gamemodes/Gamemode.cs

```diff
@@ -74,14 +80,12 @@ internal virtual void Initialize()
 
 	internal virtual void Start() { }
 
-	internal virtual void OnClientJoined( IClient client )
-	{
-		PlayerCount++;
-	}
+	internal virtual void OnClientJoined( IClient client ) { }
 
 	internal virtual void OnClientDisconnect( IClient cl, NetworkDisconnectionReason reason )
 	{
-		PlayerCount--;
+		if ( cl.Pawn is Player player )
+			DisconnectedPlayers.Add( player );

```

Comment:
Since the `Player` is only created when ffa is instantiated, players leaving and joining won't update this list. Do we want to move the implementation into `FreeForAll.cs`?

I'm unable to remove the disconnected player from `Players`/`PlayerRotation` since they get iterated over in a task (if you were wondering why I didn't do that here).

Reply:
This is fine for now. Let's take another look at it when we look into implementing more gamemodes 👍

Reply:
Yeah we should spend some time rewriting the base gamemode code.